tests/test_expectation_maximization.py

```diff
@@ -99,11 +115,37 @@ def test_generate_slices(test_ir, n_particles, n_pix):
     rots = test_ir.grid_SO3_uniform(n_particles)
     xy_plane = test_ir.generate_xy_plane(n_pix)
 
-    slices, xyz_rotated = test_ir.generate_slices(map_3d, xy_plane, n_pix, rots)
+    slices, xyz_rotated_planes = test_ir.generate_slices(map_3d, xy_plane, n_pix, rots)
 
-    assert xy_plane.shape == (n_pix**2, 3)
     assert slices.shape == (n_particles, n_pix, n_pix)
-    assert xyz_rotated.shape == (n_pix**2, 3)
+    assert xyz_rotated_planes.shape == (n_particles, 3, n_pix**2)
+
+    rot_test_map = np.array(
+        [
+            [[0, 0, 0], [0, 0, 0], [0, 0, 0]],
+            [[1, 1, 1], [1, 1, 1], [1, 1, 1]],
+            [[0, 0, 0], [0, 0, 0], [0, 0, 0]],
+        ]
+    )
+
+    rot_mat = np.array(
+        [
+            [[0, 0, 1], [0, 1, 0], [-1, 0, 0]],
+        ]
+    )
+
+    xy_plane = np.array(
+        [
+            [-1, 0, 1, -1, 0, 1, -1, 0, 1],
+            [-1, -1, -1, 0, 0, 0, 1, 1, 1],
+            [0, 0, 0, 0, 0, 0, 0, 0, 0],
+        ]
+    )
+
+    slices, xyz_rotated_planes = test_ir.generate_slices(
+        rot_test_map, xy_plane, 3, rot_mat
+    )
+    assert np.allclose(slices[0], np.ones_like(slices[0]))
```

Comment:
add some notes why this should be one. What is going on? Is the map all ones in the central slice, and we are just doing an in plane rotation? Is it ones along a different axis, and we are rotating 90 deg around the right axis to generate all ones in the plane. See my tests documentation in test_binary_mask_3d
https://github.com/compSPI/reconstructSPI/blob/dev/tests/test_expectation_maximization.py#L202

At a past meeting @fredericpoitevin mentioned having very good tests, so that we know where our axes are defined etc. Having several tests will be very helpful here.

Some ideas for tests:
* rotation of identity, map has all ones already along the central slice but zeros elsewhere. Return is ones and serves to clarify what the elements of the input correspond to the central slice
* rotation of 90 deg in the xy plane.
  * input map has ones on central slice and zeros elsewhere. return is all ones.
  * input map has simple gradient along one axis. return has gradient along other axis. helps define which was is pointing up and down along one axis. can repeat with x and y.
* rotation of zx or zy plane (puts z axis to +/- x axis or +/- y axis). input of arr_3d and output cases cases of slices that help define direction of x,y,z axis.